File: app/src/main/java/com/client/vpman/weatherwall/model/PagerAgentViewModel.java
package com.client.vpman.weatherwall.model;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class PagerAgentViewModel extends ViewModel {
    private MutableLiveData<CharSequence> text = new MutableLiveData<>();
    public void setText(CharSequence input) {
        text.setValue(input);
    }
    public LiveData<CharSequence> getText() {
        return text;
    }
}

File: app/src/main/java/com/client/vpman/weatherwall/model/PagerAgentViewModel.kt
package com.client.vpman.weatherwall.model

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class PagerAgentViewModel : ViewModel() {
    private val text = MutableLiveData<CharSequence>()
    fun setText(input: CharSequence) {
        text.value = input
    }

    fun getText(): LiveData<CharSequence> {
        return text
    }
}